r/aws_globalaccelerator_endpoint_group: Add aws_globalaccelerator_endpoint_group resource (#8328)
terraform-providers/terraform-provider-aws: website/docs/r/globalaccelerator_endpoint_group.html.markdown

---
layout: "aws"

> page_title: "AWS: globalaccelerator_endpoint_group"

Reviewer: ```suggestion
page_title: "AWS: aws_globalaccelerator_endpoint_group"
```

sidebar_current: "docs-aws-resource-globalaccelerator-endpoint-group"
description: |-
  Provides a Global Accelerator endpoint group.
---

> # Resource: globalaccelerator_endpoint_group

Reviewer: ```suggestion
# Resource: aws_globalaccelerator_endpoint_group
```

Provides a Global Accelerator endpoint group.

> ## Attributes Reference

Reviewer: The attributes section should be below the argument section.

In addition to all arguments above, the following attributes are exported:

> * `id` - The Amazon Resource Name (ARN) of the accelerator.

Reviewer: ```suggestion
* `id` - The Amazon Resource Name (ARN) of the endpoint group.
```

## Example Usage

```hcl
resource "globalaccelerator_endpoint_group" "example" {

>   listener_arn            = "arn:aws:globalaccelerator::123456789012:accelerator/1234abcd-abcd-1234-abcd-1234abcdefgh/listener/0123vxyz"

Reviewer: Examples should prefer to use resource references where possible to illustrate real world usage, e.g.

```suggestion
  listener_arn            = "${aws_globalaccelerator_listener.example.id}"
```

  endpoint_configuration {

>     EndpointId   = "arn:aws:elasticloadbalancing:us-east-1:123456789012:loadbalancer/app/my-load-balancer/50dc6c495c0c9188"

Reviewer: The argument name here is `endpoint_id`. Also, same here where examples should prefer to use resource references where possible to illustrate real world usage, e.g.

```suggestion
    endpoint_id   = "${aws_lb.example.arn}"
```

>     Weight = "100"

Reviewer: The argument name here is lowercase `weight` and is an integer which does not need to be wrapped in quotes:

```suggestion
    weight = 100
```

* `health_check_port` - (Optional) The port that AWS Global Accelerator uses to check the health of endpoints that are part of this endpoint group. The default port is the listener port that this endpoint group is associated with. If listener port is a list of ports, Global Accelerator uses the first port in the list.
* `health_check_protocol` - (Optional) The protocol that AWS Global Accelerator uses to check the health of endpoints that are part of this endpoint group. The default value is TCP.
* `threshold_count` - (Optional) The number of consecutive health checks required to set the state of a healthy endpoint to unhealthy, or to set an unhealthy endpoint to healthy. The default value is 3.
* `traffic_dial_percentage` - (Optional) The percentage of traffic to send to an AWS Region. Additional traffic is distributed to other endpoint groups for this listener. The default value is 100.
* `endpoint_configuration` - (Optional) The list of endpoint objects. Fields documented below.

**endpoint_configuration** supports the following attributes:

> * `EndpointId` - (Optional) An ID for the endpoint. If the endpoint is a Network Load Balancer or Application Load Balancer, this is the Amazon Resource Name (ARN) of the resource. If the endpoint is an Elastic IP address, this is the Elastic IP address allocation ID.

Reviewer: ```suggestion
* `endpoint_id` - (Optional) An ID for the endpoint. If the endpoint is a Network Load Balancer or Application Load Balancer, this is the Amazon Resource Name (ARN) of the resource. If the endpoint is an Elastic IP address, this is the Elastic IP address allocation ID.
```

> * `Weight` - (Optional) The weight associated with the endpoint. When you add weights to endpoints, you configure AWS Global Accelerator to route traffic based on proportions that you specify. 

Reviewer: ```suggestion
* `weight` - (Optional) The weight associated with the endpoint. When you add weights to endpoints, you configure AWS Global Accelerator to route traffic based on proportions that you specify. 
```
